tests/unit/nupic/research/temporal_memory_test.py

```diff
@@ -59,368 +68,848 @@ def testInitInvalidParams(self):
-
-
-  def testActivateCorrectlyPredictiveCellsEmpty(self):
-    tm = self.tm
-
-    # No previous predictive cells, no active columns
-    prevPredictiveCells = set()
-    activeColumns      = set()
-    prevMatchingCells = set()
-
-    (activeCells,
-    winnerCells,
-    predictedColumns,
-    predictedInactiveCells) = tm.activateCorrectlyPredictiveCells(prevPredictiveCells,
-                                                                  prevMatchingCells,
-                                                                  activeColumns)
-
-    self.assertEqual(activeCells,      set())
-    self.assertEqual(winnerCells,      set())
-    self.assertEqual(predictedColumns, set())
-    self.assertEqual(predictedInactiveCells, set())
-
-    # No previous predictive cells, with active columns
-
-    prevPredictiveCells = set()
-    activeColumns = set([32, 47, 823])
-    prevMatchingCells = set()
-
-    (activeCells,
-    winnerCells,
-    predictedColumns,
-    predictedInactiveCells) = tm.activateCorrectlyPredictiveCells(prevPredictiveCells,
-                                                                  prevMatchingCells,
-                                                                  activeColumns)
-
-    self.assertEqual(activeCells,      set())
-    self.assertEqual(winnerCells,      set())
-    self.assertEqual(predictedColumns, set())
-    self.assertEqual(predictedInactiveCells, set())
-
-    # No active columns, with previously predictive cells
-
-    prevPredictiveCells = set([0, 237, 1026, 26337, 26339, 55536])
-    activeColumns = set()
-    prevMatchingCells = set()
-
-    (activeCells,
-    winnerCells,
-    predictedColumns,
-    predictedInactiveCells) = tm.activateCorrectlyPredictiveCells(prevPredictiveCells,
-                                                                  prevMatchingCells,
-                                                                  activeColumns)
-
-    self.assertEqual(activeCells,      set())
-    self.assertEqual(winnerCells,      set())
-    self.assertEqual(predictedColumns, set())
-    self.assertEqual(predictedInactiveCells, set())
-
-  def testActivateCorrectlyPredictiveCellsOrphan(self):
-    tm = self.tm
-    tm.predictedSegmentDecrement = 0.001
-    prevPredictiveCells = set([])
-    activeColumns = set([32, 47, 823])
-    prevMatchingCells = set([32, 47])
-
-    (activeCells,
-    winnerCells,
-    predictedColumns,
-    predictedInactiveCells) = tm.activateCorrectlyPredictiveCells(prevPredictiveCells,
-                                                                  prevMatchingCells,
-                                                                  activeColumns)
-
-    self.assertEqual(activeCells, set([]))
-    self.assertEqual(winnerCells, set([]))
-    self.assertEqual(predictedColumns, set([]))
-    self.assertEqual(predictedInactiveCells, set([32,47]))
-
-  def testBurstColumns(self):
+  def testActivateCorrectlyPredictedCells(self):
```

Comment:
After a bit of confusion I did pick up on that; however, the test suite in question uses the same variable names as the c++ version to ensure that they are easily relatable to someone looking at both of them. So the test suite shouldn't have the "predicted" vs. "predictive" confusion.